Screenreader testing tweaks

nickcolley opened the pull request:
WIP just getting this up as an update.

jsimplicio commented:
What if inside each setting they had an `<h3>` that said "change password", "change username", etc. Would the screenreader catch that first?

gr2m commented on www/about.html:
```diff
@@ -24,7 +24,10 @@
     <main>
-      <a href="/">← Return to the tracker</a>
+      <a href="/">
+        <span aria-hidden="true">←</span>
```
should we maybe do this with a css pseudo `:before` thingy? I dunno, just wondering what’s better

nickcolley replied:
I think CSS content is visible by a screenreader but I'd have to try it out to be 100% sure.

gr2m replied:
I think there are CSS instructios you can add to make it invisible to screenreaders though, similar tothe `aria-` attributes. But this has been a long while since I worked on these things

jsimplicio commented:
Should you continue that fix you figured out when settings change for it to read out the title here? Take a stab :+1:

nickcolley commented:
@jsimplicio wanted to let someone else have a go before going full steam a head, will update this branch with the fix :)

nickcolley commented:
@jsimplicio headings will not be automatically read because the screenreader will only automatically read what's focused.

nickcolley commented:
I've added the announcer, I can do a screencast if anyone is having trouble setting up a screenreader but I'm using http://www.chromevox.com/ to test (not wildly used by most people but hopefully good enough here)

I've also added rudimentary routing with hashes, now when you visit a sub view of account it'll add a hash to the URL and you can reload and be at the same subview.

Approach is similar to https://github.com/hoodiehq/hoodie-app-tracker/issues/20#issuecomment-192880181

jsimplicio commented:
tell me once you updated it with the fix I can try voice over on a mac.

nickcolley commented:
Sorry, it is updated with the fix :)

jsimplicio commented:
I've never used Voice Over before, do I just enable inside Accessibility then go inside the Tracker site and it should read everything out to me?

finnp commented:
@jsimplicio You should make sure to use Safari and then there are a few things that are good to know: http://a11yproject.com/posts/getting-started-with-voiceover.

nickcolley commented:
@jsimplicio how did you get on, did it work out okay?

jsimplicio commented:
Hey @nickcolley believe so!
Every time I changed settings it would say "Change password page loaded" or "Change email page loaded". Is that what you were looking for?

nickcolley commented:
Yeah! Looks like it's working out in VoiceOver, thanks so much for trying it out. :)

jsimplicio commented:
Cool this LGTM!

nickcolley commented:
ping @hoodiehq/maintainers

christophwitzko commented:
LGTM :+1:

christophwitzko commented:
rebased :sunglasses: LGTM :+1:
